docs/DeveloperGuide.md

```diff
@@ -4,10 +4,150 @@
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+## Design
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Architecture
+![architecture diagram](images/architecture_diagram.png)
```

Comment:
The architecture diagram does not have to connection between Parser and Commands, while the command diagram indicates otherwise.
<img width="365" alt="image" src="https://user-images.githubusercontent.com/69478335/160958275-e4e1b0e9-f0b4-4255-a512-f5fa1a8a0563.png">
<img width="735" alt="image" src="https://user-images.githubusercontent.com/69478335/160958342-e56b8e78-1b49-47a2-be01-1fd2d9ecf85d.png">